### PR TITLE
Add no-crate filter option on rustdoc

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
@onur asked me about it so here it is!

r? @QuietMisdreavus

### CODE REVIEW

File: src/librustdoc/lib.rs [hunk 1]
```diff
@@ -338,6 +338,11 @@ fn opts() -> Vec<RustcOptGroup> {
                        "enable-index-page",
                        "To enable generation of the index page")
         }),
+        unstable("no-doc-crate-filtering", |o| {
+            o.optflag("",
+                      "no-doc-crate-filtering",
+                      "to prevent crate filtering button generation on search")
```
Comment (QuietMisdreavus) [inline src/librustdoc/lib.rs]: > `--disable-per-crate-search`
>
> "disables generating the crate selector on the search box"

**Review by QuietMisdreavus — COMMENTED** [review]
Just a nit on the wording of the flag, otherwise this looks good!

This is going to conflict with https://github.com/rust-lang/rust/pull/57011, so whichever merges first will leave the other with a merge conflict.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/librustdoc/config.rs [hunk 2]
```diff
@@ -192,6 +192,9 @@ pub struct RenderOptions {
     /// If present, playground URL to use in the "Run" button added to code samples generated from
     /// standalone Markdown files. If not present, `playground_url` is used.
     pub markdown_playground_url: Option<String>,
+    /// If false, the `select` element to have search filtering by crates on rendered docs
+    /// won't be generated.
+    pub generate_filter_crates: bool,
```
Comment (QuietMisdreavus) [inline src/librustdoc/config.rs]: We should also change this to something like `generate_search_filter` to match the CLI flag a little closer.

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**